### ISSUE TITLE
Write fork(trace) = deepcopy(trace) or something equivalent

### ISSUE THREAD

**Issue by marcoct** [probcomp/GenExperimental.jl#19]

**Comment by vkmvkmvkmvkm** [comment]
We may want Taylor to implement a persistent data structure here, so we
have support for copy-on-write trace forking. Does that make sense?

**Comment by marcoct** [comment]
Yes, that makes sense. I will need to think about this a bit more. In addition to forking traces for use in SIR / SMC, we have proposal programs proposing trace fragments into model programs (https://github.com/probcomp/Gen.jl/issues/1). Persistent trace data structure may also be relevant for that.
